### DISCUSSION TITLE
RavenDb Progress

### DISCUSSION THREAD

**Discussion by aboutdomtime** [abpframework/abp#10346] [Ideas]
I am currently working on integrating RavenDb into ABP. I've decided to use use the MongoDB tests as a base. So far I have all the "Repositories", "Transactions" and "UOW" tests passing.  RavenDB follows the UOW pattern natively, so integrating that with ABP was a bit of a pain. Also, RavenDB has both an IAsyncDocumentSession and a IDocumentSession which makes operations such as Repository.WithDetailsAsync(x => x.Phones).FirstOrDefault() impossible, because we're mixing async with sync.  Currently, two of the unit tests are mixing these operations, so I had to override them. Besides that everything seems to be working well.  

I plan on tackling the "DomainEvents" ,"Domain", "DataFiltering" and "Auditing" tests passing over the next two weeks.

If anyone is interested in working on this together, I will create a Fork and hopefully get to do a pull-request at some point.

**Comment by realLiangshiwei** [comment]
Hi,
We always welcome contributors,  you can send PR any time.
